### ISSUE TITLE
pypowsybl integration with Hades2 issue

### ISSUE THREAD

**Issue by hugo-antoine-rtei** [powsybl/pypowsybl#119]
* **Do you want to request a *feature* or report a *bug*?**
I want to report a possible *bug*.

* **What is the current behavior?**
The IEEE14 loadflow can't be solved using Hades2. I can't determine if it is a Hades2 issue, or a pypowsybl issue. The loadflow does not raise any exception, however, no result is generated.

* **If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem**
pypowsybl 0.7.0 was installed using the provided steps:
    - pip3 install --upgrade setuptools pip --user
    - pip3 install pypowsybl --user

    It seems to run as expected (the IEEE14 network loadflow can be computed without issues).
    But, when we add Hades2, using the given [steps](https://rte-france.github.io/hades2/index.html)
    The IEEE14 loadflow using Hades2 runs, but returns an empty result, with no exception raised.

* **What is the expected behavior?**
Raise of an exception, to allow the diagnosis of the issue, as I still don't know if it comes from pypowsybl or Hades2.

* **What is the motivation / use case for changing the behavior?**
clarity, usability

* **Please tell us about your environment:**
  - PowSyBl Version: 0.7.0
  - OS Version: Ubuntu 20.04.2
  - Hade2 Version: 6.7.1.1

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, spectrum, etc)

  - The IEEE14 loadflow without using Hades2:
```
import pypowsybl as pp
n = pp.network.create_ieee14()
results = pp.loadflow.run_ac(n)
for result in results:
    print(result)
print(results)
print(len(results))
```
```
LoadFlowComponentResult(component_num=0, status=CONVERGED, iteration_count=3, slack_bus_id='VL4_0', slack_bus_active_power_mismatch=-0.0060808596928974)
<_pypowsybl.LoadFlowComponentResultArray object at 0x7f86b958c430>
1
```
-
  - The IEEE14 loadflow with Hades2:
```
import pypowsybl as pp
n = pp.network.create_ieee14()
results = pp.loadflow.run_ac(n, provider="Hades2")
for result in results:
    print(result)
print(results)
print(len(results))
```
```

<_pypowsybl.LoadFlowComponentResultArray object at 0x7f3b8285eb30>
0
```

I don't know if this is related, but I noticed that I could run the input example given in the Hades2 archive using the .sh:

`./hades2.sh LF input_IEEE14nodes.xml output_test.xml`
>RTE load flow tool - RTE HADES trial and academic license expiring the 01/01/2022
#TP#Execution sur 1 thread(s)
#TP#HADES2 : START
#TP#Computing networg parameters initialization
#TP#Building of computing network
#TP#HADES2 : END connected component num 1
#TP#HADES2 : END

But that I couldn't run the input example given in the Hades2 archive using the executable in the /bin folder:

`./hades2 LF input_IEEE14nodes.xml output_test2.xml`
>./hades2: error while loading shared libraries: libxerces-c.so.27: cannot open shared object file: No such file or directory

So, I assume it might be linked to a library I would be missing. However I couldn't find where I could download and install it. And I wanted to try and ask for some insight, before messing something up.

**Comment by geofjamg** [comment]
Hi,
This is an issue related to hades2 integration in powsybl (not directly an issue of pypowsybl). Part of the loadflow result is not implemented in hades2 integration. Any ways calculation is done and state is correctly set in the model after calculation (voltage and power flows).
I forward this issue to people responsible of the component for having a fix.

**Comment by jonenst** [comment]
Also, the shared library problem that you see when running the binary directly is normal, that's exactly what hades2.sh is for: setting up LD_LIBRARY_PATH so that the linker can find the library shipped in the hades2 distribution ( libxerces-c.so.27 is in the lib folder)

**Comment by hugo-antoine-rtei** [comment]
Hello everyone,
Thanks a lot for the answers !
Thank you @jonenst for pointing out I was on the wrong track, I would have wasted a lot of time if you didn't.
@geofjamg, is there already an issue created that I should be watching ? Or should I just watch this one ?

**Comment by geofjamg** [comment]
Solved in 0.9.0 thanks to hades2 integration fix

**Comment by geofjamg** [comment]
Solved in 0.9.0 thanks to hades2 integration fix
